ocaml/runtime/minor_gc.c

```diff
@@ -308,13 +308,20 @@ static void oldify_one (void* st_v, value v, volatile value *p)
     st->live_bytes += Bhsize_hd(hd);
     result = alloc_shared(st->domain, sz, tag, Reserved_hd(hd));
     field0 = Field(v, 0);
+    mlsize_t scannable_sz = Scannable_wosize_hd(hd);
+    /* Copy the non-scannable suffix of fields. */
+    for (i = scannable_sz; i < sz; i++) {
+      Field(result, i) = Field(v, i);
+    }
```

Comment:
For multicore performance, this loop should be below the `if (try_update_object_header ...)`.

With multiple domains, if a block is reachable from multiple domains during minor GC then both will race to promote it, and `try_update_object_header` will return true only on the domain that wins the race. So, moving the loop down helps in two ways:

  - you are more likely to win the race if you don't stop to do a memcpy before crossing the finish line
  - if you lose the race, you don't need to do the copy at all

Reply:
I accepted this suggestion.

Reply:
(My naive attempt to move the `memcpy` down introduced a bug, because I forgot that `try_update_object_header` clobbers the 0th field of `v`. I've just pushed a fix.)

Reply:
Well spotted on the bug, I hadn't thought of that!